libnetwork/drivers/macvlan/macvlan_endpoint.go

```diff
@@ -30,11 +31,14 @@ func (d *driver) CreateEndpoint(ctx context.Context, nid, eid string, ifInfo dri
 		addrv6: ifInfo.AddressIPv6(),
 		mac:    ifInfo.MacAddress(),
 	}
-	if ep.addr == nil {
-		return fmt.Errorf("create endpoint was not passed an IP address")
-	}
 	if ep.mac == nil {
-		ep.mac = netutils.GenerateMACFromIP(ep.addr.IP)
+		if ep.addr != nil {
+			ep.mac = netutils.GenerateMACFromIP(ep.addr.IP)
+		} else {
+			// TODO(robmry) - generate an unsolicited ARP to associate this MAC
```

Comment:
ARP? Do you mean IPv6 Neighbor Advertisement? It doesn't make much sense to send an unsolicited ARP announcement without an IPv4 address to announce.

Reply:
Yes, fixed.